data/data/azure/master/master.tf

```diff
@@ -0,0 +1,102 @@
+resource "azurerm_network_interface" "master" {
+  count               = "${var.instance_count}"
+  name                = "${var.cluster_id}-master-nic-${count.index}"
+  location            = "${var.region}"
+  resource_group_name = "${var.resource_group_name}"
+
+  ip_configuration {
+    subnet_id                     = "${var.subnet_id}"
+    name                          = "master-${count.index}"
+    private_ip_address_allocation = "Dynamic"
+  }
+}
+
+resource "azurerm_network_interface_backend_address_pool_association" "master" {
+  count                   = "${var.instance_count}"
+  network_interface_id    = "${element(azurerm_network_interface.master.*.id, count.index)}"
+  backend_address_pool_id = "${var.elb_backend_pool_id}"
+  ip_configuration_name   = "master-${count.index}"                                          #must be the same as nic's ip configuration name.
+}
+
+resource "azurerm_network_interface_backend_address_pool_association" "master_internal" {
+  count                   = "${var.instance_count}"
+  network_interface_id    = "${element(azurerm_network_interface.master.*.id, count.index)}"
+  backend_address_pool_id = "${var.ilb_backend_pool_id}"
+  ip_configuration_name   = "master-${count.index}"                                          #must be the same as nic's ip configuration name.
+}
+
+resource "azurerm_lb_nat_rule" "master" {
+  resource_group_name            = "${var.resource_group_name}"
+  enable_floating_ip             = false
+  loadbalancer_id                = "${var.external_lb_id}"
+  name                           = "natRuleSsh${count.index}"
+  protocol                       = "Tcp"
+  frontend_port                  = "${2200 + count.index}"
+  backend_port                   = 22
+  frontend_ip_configuration_name = "PublicIPAddress"
+  count                          = "${var.instance_count}"
+}
+
+resource "azurerm_network_interface_nat_rule_association" "master" {
+  count                 = "${var.instance_count}"
+  network_interface_id  = "${element(azurerm_network_interface.master.*.id, count.index)}"
+  ip_configuration_name = "master-${count.index}"
+  nat_rule_id           = "${element(azurerm_lb_nat_rule.master.*.id, count.index)}"
+}
+
+#TODO : make FD/UD configurable
+resource "azurerm_availability_set" "master" {
```

Comment:
This needs more information as to why we are using this and why machine-api doesn't solve this requirement?

Reply:
this is how Azure manages update domain and fault domain. Machine API has no knowledge of how the azure datacenters are organize AFAIK. This will also likely change when we add support for availibility zones here.